harness/determined/exec/harness.py

```diff
@@ -138,20 +138,7 @@ def build_and_run_training_pipeline(env: det.EnvContext) -> None:
                 if env.experiment_config.debug_enabled():
                     faulthandler.dump_traceback_later(30, repeat=True)
 
-                (
-                    prof_start_on_batch,
-                    prof_end_after_batch,
-                ) = env.experiment_config.profiling_interval()
-                with profiler.ProfilerAgent(
-                    trial_id=env.det_trial_id,
-                    agent_id=env.det_agent_id,
-                    master_url=env.master_url,
-                    profiling_is_enabled=env.experiment_config.profiling_enabled(),
-                    global_rank=0,
-                    local_rank=0,
-                    start_on_batch=prof_start_on_batch,
-                    end_after_batch=prof_end_after_batch,
-                ) as prof:
+                with profiler.ProfilerAgent.from_env(env, global_rank=0, local_rank=0) as prof:
```

Comment:
What is the relationship between this PR and #2355?  It looks like this is editing what #2355 is adding, but I only see one commit on this PR.

Reply:
This PR is merging into the `hook_up_sysmetric_profiler` feature branch (PR #2355). This is on top of #2355